<launch>
	<arg name="sim" default="false"/>
	<arg name="run_move_gripper_server" default="true"/>
	<arg name="run_grasps_server" default="true"/>
	<arg name="run_take_pose_server" default="true"/>
	<arg name="run_place_server" default="true"/>
	<arg name="run_open_server" default="true"/>
	<arg name="run_open_drawer_server" default="true"/>
	<arg name="run_make_plan_server" default="true"/>
	<arg name="run_object_state_listener" default="true"/>
	<arg name="run_manipulation_test_marker" default="true"/>
	<arg name="load_static_objects" default="true"/>
    <arg name="gui" default="false"/>

	<rosparam ns="/manipulation" file="$(find suturo_manipulation)/params/hsr_params.yaml" />

	<group if="$(eval sim == true)">
		<include file="$(find iai_hsr_sim)/launch/ros_control_sim.launch" />
		<include file="$(find giskardpy)/launch/giskardpy_hsr.launch" >
			<arg name="sim" value="true"/>
			

		</include>
		<node pkg="rviz" type="rviz" name="rviz" args="-d $(find suturo_manipulation)/config/iai_hsr_sim.rviz" />
	</group>
	
	<group if="$(eval sim == false)">
		<include file="$(find giskardpy)/launch/giskardpy_hsr.launch" >
			<arg name="sim" value="false"/>
			

		</include>
	</group>	

	<node pkg="suturo_manipulation" name="object_state_listener" type="object_state_listener" output="screen"
		  if="$(eval run_object_state_listener == true)">
		<remap from="~object_state" to="/object_state"/>
	</node>

	<!--node pkg="suturo_manipulation" type="giskard_load_static_objects.py" name="load_static_objects" if="$(eval load_static_objects == true)">
		<param name="environment_frame" value="iai_kitchen/world"/>
	</node-->

    <node pkg="giskardpy" type="add_urdf.py" name="add_kitchen">
		<param name="name" value="iai_kitchen"/>
		<param name="param" value="hsrb_lab"/>
		<param name="js" value="kitchen/joint_states"/>

		<param name="root_frame" value="iai_kitchen/urdf_main"/>
		<param name="frame_id" value="map" />
    </node>

	<node pkg="suturo_manipulation" name="move_gripper_server" type="move_gripper_server.py" output="screen"
		if="$(eval run_move_gripper_server == true)"/>

	<node pkg="suturo_manipulation" name="grasp_server" type="grasp_object_server.py" output="screen"
		if="$(eval run_grasps_server == true)"/>

	<node pkg="suturo_manipulation" name="place_server" type="place_object_server.py" output="screen"
		if="$(eval run_place_server == true)"/>

	<node pkg="suturo_manipulation" name="make_plan_server" type="make_plan_server.py" output="screen"
		if="$(eval run_make_plan_server == true)"/>

	<node pkg="suturo_manipulation" name="take_pose_server" type="take_pose_server.py" output="screen"
		if="$(eval run_take_pose_server == true)"/>

	<node pkg="suturo_manipulation" name="open_server" type="open_server.py" output="screen"
		if="$(eval run_open_server == true)"/>

	<!--<node pkg="suturo_manipulation" name="open_drawer_server" type="open_drawer_server.py" output="screen"
		if="$(eval run_open_drawer_server == true)"/>-->



	<node pkg="suturo_manipulation" name="manipulation_test_marker" type="manipulation_test_marker.py" output="screen"
		if="$(eval run_manipulation_test_marker == true)"/>

</launch>
